src/modules/link/graphql/types.rs

```diff
@@ -39,3 +39,24 @@ impl From<LinkModel> for Link {
         }
     }
 }
+
+pub struct TypeGQL<T>(pub T);*/
+
+#[async_graphql::Object(name = "Link")]
+impl TypeGQL<Link> {
+    async fn id(&self) -> String {
+        from_utf8(&self.0.id).unwrap().to_string()
+    }
+
+    async fn original_url(&self) -> &String {
+        &self.0.original_url
+    }
+
+    async fn created_at(&self) -> DateTime<Local> {
+        self.0.created_at
+    }
+
+    async fn updated_at(&self) -> DateTime<Local> {
+        self.0.updated_at
+    }
+}
```

Comment:
Why not use `SimpleObject` directly as provided by the crate? 🤔

https://async-graphql.github.io/async-graphql/en/define_simple_object.html#simpleobject

For default/linear resolvers which returns the field from the type (without computations) its easier and straightforward to use `SimpleObject` which maps the inner type field to a resolver.

This way we don't have to write a resolver for each field manually, which is error prone and has extra maintenance cost. 

Then if we wanted to write a custom resolver for another field, we could just move into `ComplexObject` as its done here:

https://async-graphql.github.io/async-graphql/en/define_simple_object.html#user-defined-resolvers

Theres an example in this repo for the "User's Links" use case:

https://github.com/whizzes/linx/blob/04931ed8af944ebf1ced867345de3adbddc21ad4/src/modules/user/graphql/types.rs#L43

Reply:
Personally, I think it's cleaner, you no longer have to use FROM for the conversion, you don't have to deal with types that have the same name as the models and be using aliases, the graphql types are linked to the models and not entities independent, I do not think they are prone to errors on the contrary they prevent it.

It's my point of view, I return it as it was 😊